pkg/storage/replica_command.go

```diff
@@ -2005,11 +2005,22 @@ func (r *Replica) sha512(
 	desc roachpb.RangeDescriptor, snap engine.Reader, snapshot *roachpb.RaftSnapshotData,
 ) ([]byte, error) {
 	hasher := sha512.New()
+	tombstoneKey := engine.MakeMVCCMetadataKey(keys.RaftTombstoneKey(desc.RangeID))
+
 	// Iterate over all the data in the range.
 	iter := NewReplicaDataIterator(&desc, snap, true /* replicatedOnly */)
 	defer iter.Close()
+
 	for ; iter.Valid(); iter.Next() {
 		key := iter.Key()
+		if key.Equal(tombstoneKey) {
+			// Skip the tombstone key which is marked as replicated even though it
+			// isn't.
+			//
+			// TODO(peter): Figure out a way to migrate this key to the unreplicated
```

Comment:
Did you file an issue for this? Seems problematic and worth fixing sooner than a TODO in the code is likely to prompt.